tests/NHapi.NUnit/PipeParsingFixture271.cs

```diff
@@ -1,168 +1,170 @@
+            bool parsedCorrectNumberOfObservations = parsedObservations == expectedObservationCount;
+            Assert.IsTrue(
+                parsedCorrectNumberOfObservations,
+                string.Format("Expected {1} OBX repetitions used for this segment, found {0}", parsedObservations, expectedObservationCount));
+
+            int index = 0;
+            var obs = oru.GetPATIENT_RESULT(0).GetORDER_OBSERVATION(0).GetOBSERVATION(index).OBX.GetObservationValue().FirstOrDefault();
+            Assert.IsTrue(obs.Data is DT);
+            index++;
+            obs = oru.GetPATIENT_RESULT(0).GetORDER_OBSERVATION(0).GetOBSERVATION(index).OBX.GetObservationValue().FirstOrDefault();
+            Assert.IsTrue(obs.Data is ST);
+            index++;
+            obs = oru.GetPATIENT_RESULT(0).GetORDER_OBSERVATION(0).GetOBSERVATION(index).OBX.GetObservationValue().FirstOrDefault();
+            Assert.IsTrue(obs.Data is TM);
+        }
+
+        [Test]
+        public void TestADTA04IsMappedAsA01()
+        {
+            string hl7Data = @"MSH|^~\&|CohieCentral|COHIE|Clinical Data Provider|TCH|20060228155525||ADT^A04|1|P|2.7.1|
 EVN|
 PID|1|12345
 PV1|1".Replace(Environment.NewLine, "\r");
 
-			PipeParser parser = new PipeParser();
-			IMessage msg = parser.Parse(hl7Data);
-
-			Assert.IsNotNull(msg, "Message should not be null");
-			ADT_A01 a04 = (ADT_A01)msg;
-
-			Assert.AreEqual("A04", a04.MSH.MessageType.TriggerEvent.Value);
-			Assert.AreEqual("1", a04.PID.SetIDPID.Value);
-		}
-
-		[Test]
-		public void TestAdtA04AndA01MessageStructure()
-		{
-			var result = PipeParser.GetMessageStructureForEvent("ADT_A04", "2.7.1");
-			bool isSame = string.Compare("ADT_A01", result, StringComparison.InvariantCultureIgnoreCase) == 0;
-			Assert.IsTrue(isSame, "ADT_A04 returns ADT_A01");
-
-			result = PipeParser.GetMessageStructureForEvent("ADT_A13", "2.7.1");
-			isSame = string.Compare("ADT_A01", result, StringComparison.InvariantCultureIgnoreCase) == 0;
-			Assert.IsTrue(isSame, "ADT_A13 returns ADT_A01");
-
-			result = PipeParser.GetMessageStructureForEvent("ADT_A08", "2.7.1");
-			isSame = string.Compare("ADT_A01", result, StringComparison.InvariantCultureIgnoreCase) == 0;
-			Assert.IsTrue(isSame, "ADT_A08 returns ADT_A01");
-
-			result = PipeParser.GetMessageStructureForEvent("ADT_A01", "2.7.1");
-			isSame = string.Compare("ADT_A01", result, StringComparison.InvariantCultureIgnoreCase) == 0;
-			Assert.IsTrue(isSame, "ADT_A01 returns ADT_A01");
-		}
-
-		/// <summary>
-		/// https://github.com/nHapiNET/nHapi/issues/135
-		/// </summary>
-		[TestCaseSource(nameof(_validV271ValueTypes))]
-		public void TestObx5DataTypeIsSetFromObx2_AndAllDataTypesAreConstructable(Type expectedObservationValueType)
-		{
-			var message = $@"MSH|^~\&|XPress Arrival||||200610120839||ORU^R01|EBzH1711114101206|P|2.7.1|||AL|||ASCII
+            PipeParser parser = new PipeParser();
```

Comment:
did you turn off the prefer var for tests?

Reply:
no - but might not have fixed these

Reply:
fixed in the whole project